metricbeat/helper/sql/sql.go

```diff
@@ -167,24 +167,25 @@ func ReplaceUnderscores(ms mapstr.M) mapstr.M {
 }
 
 func getValue(pval *interface{}) interface{} {
+	if pval == nil {
+		return nil
+	}
+
 	switch v := (*pval).(type) {
-	case nil, bool:
+	case nil, bool, int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64, string, []interface{}:
 		return v
 	case []byte:
-		s := string(v)
-		num, err := strconv.ParseFloat(s, 64)
```

Comment:
For something like below: 
data := []byte("123.45")

Should data be a string or numeric. 
In previous code we were changing to float, now it will always be a string. 
What is the expectation of the SQL module for this ?

Reply:
```
integer types smallint, integer, and bigint are returned as int64
floating-point types real and double precision are returned as float64
character types char, varchar, and text are returned as string
temporal types date, time, timetz, timestamp, and timestamptz are returned as time.Time
the boolean type is returned as bool
the bytes type is returned as []byte
```

This is how SQL types are converted to Go types using the library for Postgres.

If you see if someone has set a type as BYTES in SQL, there's expectation to keep it as []byte or string. I am not sure why earlier there was a decision to convert it to float as this as issues:

- If the query returns []byte equivalent of "00100"; converting it to float is possible and it would truncate the leading 0s. And no way to get it back.

I think it's better to keep it as string and not change it to float from the code. If someone needs to convert it to float then they can use ingest pipelines; which is doable but not the other way around i.e., float to string.

In our case, for which the bug was filed. field was a VARCHAR and Postgres client library converts it to Go's string type; and it should be kept string only (no auto assumption and conversion should be done, imo). If required, they can always do it in their custom ingest pipelines.

Reply:
This could change the behaviour for the users using the module for [ ] byte datatype. As they might be expecting numeric and might have added their own customizations per that. 

But, IMO also, its ok to have it as string. 
Just considering it should not break for the existing users.